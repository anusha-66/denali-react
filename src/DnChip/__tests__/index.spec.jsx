/**
 * Copyright 2020, Verizon Media
 * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
 */

import React from 'react';
import { mount, render } from 'enzyme';
import { DnChip, DnChipSizes, DnChipColors, DnChipShades } from '../../index'; // src/index.tsx

describe('DnChip Tests', () => {
    it('should export DnChipSizes', () => {
        expect.assertions(1);

        expect(DnChipSizes).toStrictEqual({
            default: 'default',
            large: 'large',
            medium: 'medium',
            small: 'small',
        });
    });

    it('should export DnChipColors', () => {
        expect.assertions(1);

        expect(DnChipColors).toStrictEqual({
            brand: 'brand',
            grey: 'grey',
            red: 'red',
            orange: 'orange',
            yellow: 'yellow',
            lime: 'lime',
            green: 'green',
            sky: 'sky',
            blue: 'blue',
            purple: 'purple',
            violet: 'violet',
            pink: 'pink',
        });
    });

    it('should export DnChipShades', () => {
        expect.assertions(1);

        expect(DnChipShades).toStrictEqual({
            shade100: '100',
            shade200: '200',
            shade300: '300',
            shade400: '400',
            shade500: '500',
            shade600: '600',
            shade700: '700',
            shade800: '800',
        });
    });

    it('should render with default props', () => {
        expect.assertions(1);

        const rendered = render(<DnChip>Online</DnChip>);

        expect(rendered).toMatchSnapshot();
    });

    it('should render correctly with custom props', () => {
        expect.assertions(1);

        const rendered = render(
            <DnChip
                size={DnChipSizes.large}
                bgColor={DnChipColors.red}
                bgShade={DnChipShades.shade800}
                color={DnChipColors.yellow}
                shade={DnChipShades.shade600}
                className="p-18"
            >
                Offline
            </DnChip>,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should call on click correctly', () => {
        expect.assertions(1);

        const mockedOnClick = jest.fn();
        const wrapper = mount(<DnChip onClick={mockedOnClick}>Info</DnChip>);
        wrapper.find('span.chips').simulate('click');

        expect(mockedOnClick).toHaveBeenCalledTimes(1);
    });
});
