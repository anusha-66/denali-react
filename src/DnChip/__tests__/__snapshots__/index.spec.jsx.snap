// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DnChip Tests should render correctly with custom props 1`] = `
<span
  class="chips is-large has-bg-red-800 is-yellow-600 p-18"
>
  Offline
</span>
`;

exports[`DnChip Tests should render with default props 1`] = `
<span
  class="chips is-default"
>
  Online
</span>
`;
